### PR TITLE
MOB-9179: Support nested field types

### PR DESCRIPTION
**Pull request by megha-iterable**
## Jira Ticket(s) if any

* [MOB-9179](https://iterable.atlassian.net/browse/MOB-9179)

[MOB-9179]: https://iterable.atlassian.net/browse/MOB-9179?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: iterableapi/src/androidTest/java/com/iterable/iterableapi/util/NestedCriteriaMatchTest.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+                "                  \"furnitureColor\": \"Gray\",\n" +
+                "                  \"lengthInches\": 20,\n" +
+                "                  \"widthInches\": 30\n" +
+                "                }\n" +
+                "              ]\n" +
+                "            },\n" +
+                "            \"eventType\": \"user\"\n" +
+                "          }\n" +
+                "        ]";
+        JSONArray jsonArray = new JSONArray(jsonString);
```
Comment (evantk91) [inline iterableapi/src/androidTest/java/com/iterable/iterableapi/util/NestedCriteriaMatchTest.java:125]: Based on the ticket, this scenario should pass.

Comment (evantk91) [reply iterableapi/src/androidTest/java/com/iterable/iterableapi/util/NestedCriteriaMatchTest.java:125]: We should want the scenario of two furniture
furnitureType: "Table", furnitureColor: "White" and furnitureType: "Table", furnitureColor: "Gray" to fail.

**Review by evantk91 — COMMENTED** [review]
Left some comments.

File: iterableapi/src/androidTest/java/com/iterable/iterableapi/util/NestedCriteriaMatchTest.java (hunk 1 above)
Comment (megha-iterable) [reply iterableapi/src/androidTest/java/com/iterable/iterableapi/util/NestedCriteriaMatchTest.java:125]: Update the logic. Please review it.

**Review by evantk91 — APPROVED** [review]
LGTM. Adding a comment for future refactoring.

**Pull request merged**